src/hooks/isProjectAllowed.js

```diff
@@ -65,7 +65,34 @@ const checkOwner = context => {
   return context;
 };
 
-module.exports = () => context => {
-  checkOwner(context);
-  checkReviewer(context);
+const checkToken = context => {
+  const tokenWhitelist = context.app.get('tokenWhitelist');
+
+  const items = commons.getItems(context);
+
+  const inWhitelist = project => {
+    if (tokenWhitelist.find(t => t.address === project.token.address)) return;
+
+    throw new errors.BadRequest(
+      `token ${project.token.symbol} is not in the whitelist`,
+    );
+  };
+
+  if (Array.isArray(items)) {
+    items.forEach(inWhitelist);
+  } else {
+    inWhitelist(items);
+  }
+  return context;
+};
+
+
+module.exports = {
+  isProjectAllowed: () => context => {
```

Comment:
this should be called `isMilestoneAllowed` b/c a campaign is a project as well & has no need for a token check.

Also, why do we care if a project has a non-whitelisted token? Will it break things? Seems like it shouldn't matter & this hook is really only for the reviewers/owners